compiler/rustc_middle/src/ty/predicate/to_predicate.rs

```diff
@@ -0,0 +1,103 @@
+use crate::ty::{
```

Comment:
I still think that this should live in predicates. These are essentially just `Into` implementations for `Predicate`, and it's easier to just have them live with the types they're defined on.

Reply:
I feel like we ought to just not have any of the `predicate/` files and isntead put it all in `predicate.rs` there's barely anything in the submodules and they would fit in fine in `predicate.rs`

Reply:
@compiler-errors Most of those are not implemented for `Predicate`s though, so I'm not sure I follow...

@BoxyUwU I don't agree, having the same points as before: adding more entities to the same file makes it harder to add something there (it's easy to mess with the order) and harder to navigate... I can understand the reasoning behind having all predicate parts in a single file (after all, it's essentially a single thing split into multiple types), but I can't see the point in merging `ToPredicate`, `InstantiatedPredicates` and `CratePredicatesMap` there too.